raiden/tests/utils/ci.py

```diff
@@ -1,12 +1,37 @@
 import os
+from typing import Optional
 
-from raiden.utils.typing import Optional
+from raiden.utils import pex
 
 
-def get_artifacts_storage(*parts) -> Optional[str]:
-    artifact_dir = os.environ.get("RAIDEN_TESTS_ETH_LOGSDIR")
+def get_artifacts_storage() -> Optional[str]:
+    return os.environ.get("RAIDEN_TESTS_LOGSDIR")
 
-    if artifact_dir:
-        return os.path.join(artifact_dir, *parts)
 
-    return None
+def shortened_artifacts_storage(test_node, *parts: str) -> Optional[str]:
+    """Return a pathname based on the test details.
+
+    Some platforms have a limit to the length of a file path. This function
+    will compute a path name based on the test details, and if necessary trim
+    it down to fit 300 characters.
+    """
+    artifacts_dir = get_artifacts_storage()
+
+    if artifacts_dir is None:
+        return None
+
+    path = os.path.join(artifacts_dir, test_node.name, *parts)
+
+    # Paths longer than 286 will be reject on CircleCI
+    if len(path) >= 286:
+        original_name = test_node.originalname
```

Comment:
Question: https://docs.pytest.org/en/latest/_modules/_pytest/nodes.html#Node doesnt seem to be documented. or should this be `name`?

Reply:
name is used in the line above, and it may result in a line too large, this function exists to trim the path when that is the case.